app/ts/util/swapUtil.ts

```diff
@@ -0,0 +1,13 @@
+export function getNextId<T extends {id: string}>(
+  index: number,
+  items: T[]
+): string | undefined {
+  return index < items.length - 1 ? items[index + 1].id : undefined;
+}
+
+export function getPreviousId<T extends {id: string}>(
```

Comment:
not necessarily better but something like 
```
  function getNullSafeId<T extends {id: string}>( item: T) : string | undefined {
    return item? item.id : undefined;
  }
``` 
could be more generally applicable and means we don't need separate previous and next fns

Reply:
Doesn't this get us into trouble when index === 0 and it get the previous item? array[-1] throws an error right?

Reply:
nah, no error
```
>>  [1,2,3][-1]
-> undefined
```